processing-service-core/api/models/tenant.js

```diff
@@ -0,0 +1,340 @@
+    tenantConfigStoreModel.getTenantConfig(tenantId, key)
+            .then(function (tenantConfigResult) {
+                effectiveConfig.value = tenantConfigResult;
+                effectiveConfig.valueType = "CUSTOM";
+                tenantConfig.resolve(effectiveConfig);
+            })
+            .fail(function (errorResponse) {
+                logger.debug("No tenant specific config for tenant " + tenantId + " with key " + key + " was found, returning default value");
+                if (errorResponse instanceof ApiError && errorResponse.type === apiErrorTypes.ITEM_NOT_FOUND) {
+                    globalConfigStoreModel.getGlobalConfig(tenantId, key)
+                            .then(function (tenantDefaultConfig) {
+                                for (var key in tenantDefaultConfig) {
+                                    effectiveConfig.value = tenantDefaultConfig[key];
+                                    effectiveConfig.valueType = "DEFAULT";
+                                    tenantConfig.resolve(effectiveConfig);
+                                }
+                            })
+                            .fail(function (errorResponse) {
+                                tenantConfig.reject(errorResponse);
+                            })
+                            .done();
+                } else {
+                    tenantConfig.reject(errorResponse);
+                }
+            })
+            .done();
+    return tenantConfig.promise;
+}
+
+/**
+ * Retrieve effective tenant specific config.
+ * Returns the effective tenant specific config for this key. That can be either a custom value specified for the tenant or the default.
+ *
+ * @param {String} tenantId String The ID tenant.
+ * returns effectiveTenantConfigs
+ **/
+function getEffectiveTenantConfigs(tenantId) {
+
+    var tenantConfig = Q.defer();
+    logger.debug("Retrieving efective tenant config for tenant " + tenantId);
+    var effectiveConfigs = {};
+    effectiveConfigs.tenantId = tenantId;
+    effectiveConfigs.configurations = [];
+    tenantConfigStoreModel.getTenantConfigs(tenantId)
+            .then(function (tenantConfigurations) {
+                var tenantConfigMap = createMap(tenantConfigurations);
+                for (var key in tenantConfigMap) {
+                    var config = {
+                        tenantId: tenantId,
+                        key: key,
+                        value: tenantConfigMap[key],
+                        valueType: "CUSTOM"
+                    };
+                    effectiveConfigs.configurations.push(config);
+                }
+            })
+            .fail(function (errorResponse) {
+                //Not throwing the error as this can happen when a tenant has not set any custom configurations.
+                logger.debug("Failed to find any tenant specific configs for tenant " + tenantId);
```

Comment:
This may have been a database connection failure you shouldn't just silently log it. You need to differentiate between ITEM_NOT_FOUND error types and other kinds of failures.